lib/mayaUsd/ufe/UsdConnectionHandler.cpp

```diff
@@ -88,30 +88,96 @@ bool UsdConnectionHandler::createConnection(
+        = UsdShadeUtils::GetBaseNameAndType(TfToken(dstAttr->name()));
+
+    if (srcAttrType == UsdShadeAttributeType::Input) {
+        UsdShadeInput srcInput = srcApi.CreateInput(srcBaseName, srcUsdAttr->usdAttributeType());
+        if (dstAttrType == UsdShadeAttributeType::Input) {
+            UsdShadeInput dstInput
+                = dstApi.CreateInput(dstBaseName, dstUsdAttr->usdAttributeType());
+            return UsdShadeConnectableAPI::ConnectToSource(dstInput, srcInput);
+        } else {
+            UsdShadeOutput dstOutput
+                = dstApi.CreateOutput(dstBaseName, dstUsdAttr->usdAttributeType());
+            return UsdShadeConnectableAPI::ConnectToSource(dstOutput, srcInput);
+        }
+    } else {
+        UsdShadeOutput srcOutput = srcApi.CreateOutput(srcBaseName, srcUsdAttr->usdAttributeType());
+        if (dstAttrType == UsdShadeAttributeType::Input) {
+            UsdShadeInput dstInput
+                = dstApi.CreateInput(dstBaseName, dstUsdAttr->usdAttributeType());
+            return UsdShadeConnectableAPI::ConnectToSource(dstInput, srcOutput);
+        } else {
+            UsdShadeOutput   dstOutput;
+            UsdShadeMaterial dstMaterial(dstUsdAttr->usdPrim());
+            // Special case when connecting to Material outputs:
+            if (dstMaterial
+                && (dstBaseName == UsdShadeTokens->surface || dstBaseName == UsdShadeTokens->volume
+                    || dstBaseName == UsdShadeTokens->displacement)) {
+                // Create the required output based on the type of the shader node we are trying to
+                // connect:
+                UsdShadeShader       srcShader(srcUsdAttr->usdPrim());
+                PXR_NS::SdrRegistry& registry = PXR_NS::SdrRegistry::GetInstance();
+                PXR_NS::TfToken      srcInfoId;
+                srcShader.GetIdAttr().Get(&srcInfoId);
+                PXR_NS::SdrShaderNodeConstPtr srcShaderNodeDef
+                    = registry.GetShaderNodeByIdentifier(srcInfoId);
+                if (!srcShaderNodeDef) {
+                    return false;
+                }
+                TfToken renderContext = srcShaderNodeDef->GetSourceType() == "glslfx"
+                    ? UsdShadeTokens->universalRenderContext
+                    : srcShaderNodeDef->GetSourceType();
+                if (dstBaseName == UsdShadeTokens->surface) {
+                    dstOutput = dstMaterial.CreateSurfaceOutput(renderContext);
+                } else if (dstBaseName == UsdShadeTokens->volume) {
+                    dstOutput = dstMaterial.CreateVolumeOutput(renderContext);
+                } else {
+                    dstOutput = dstMaterial.CreateDisplacementOutput(renderContext);
+                }
+            } else {
+                dstOutput = dstApi.CreateOutput(dstBaseName, dstUsdAttr->usdAttributeType());
+            }
+            return UsdShadeConnectableAPI::ConnectToSource(dstOutput, srcOutput);
```

Comment:
[Question] The previous code was using `dstUsdAttr.AddConnection()`, and the new implementation uses `UsdShadeConnectableAPI::ConnectToSource()`, why?

Reply:
I think you should convert this review comment into a code comment.